### PR TITLE
#{assert || rufute}_validates_uniquenesss_of methods

### PR DESCRIPTION
**Pull request by nessamurmur**

### CODE REVIEW

File: lib/minitest/assertions.rb [hunk 1]
```diff
@@ -35,7 +35,21 @@ def refute_validates_attachment_presence_of(object, property)
...
+  def refute_validates_uniqueness_of(object, property)
+    assert_valid(object)
+    object2 = object.clone
+    assert_valid(object2)
+    assert_nil object2.errors[property.to_sym]
+  end
+
+  def assert_valid(object)
     assert object.valid?, "Expected object to be valid"
   end
```
Comment (nessamurmur) [inline lib/minitest/assertions.rb:55]: Sorry for the weird diff... Was rebasing and didn't notice that the updated master branch had removed the parens.

File: test/lib/minitest/assertions_test.rb [hunk 2]
```diff
@@ -10,6 +10,12 @@ def setup
     @assertion_count = 1
   end
 
+  def invalid_clone(object)
+    clone = object.class.new(:property)
+    clone.property = nil
+    clone
+  end
+
```
Comment (nessamurmur) [inline test/lib/minitest/assertions_test.rb:18]: Kinda hacky but it works.

File: lib/minitest/assertions.rb [hunk 3]
```diff
@@ -35,7 +35,21 @@ def refute_validates_attachment_presence_of(object, property)
     object.send("#{property}_file_name=", tmp)
   end
 
-  def assert_valid object
+  def assert_validates_uniqueness_of(object, property)
+    assert_valid(object)
+    object2 = object.clone
+    refute object2.valid?, "Expected object not to be valid"
```
Comment (jjbohn) [inline lib/minitest/assertions.rb:41]: use `refute_valid object`

File: lib/minitest/assertions.rb [hunk 4]
```diff
@@ -35,7 +35,21 @@ def refute_validates_attachment_presence_of(object, property)
...
+    assert object2.errors[property.to_sym], "Expected errors to exist for #{property}"
```
Comment (jjbohn) [inline lib/minitest/assertions.rb]: use `refute_nil object2.errors[property.to_sym]`

File: lib/minitest/assertions.rb [hunk 5]
```diff
@@ -35,7 +35,21 @@ def refute_validates_attachment_presence_of(object, property)
...
+  end
+
+  def refute_validates_uniqueness_of(object, property)
+    assert_valid(object)
```
Comment (jjbohn) [inline lib/minitest/assertions.rb]: Could you remove the params on calls like this?

File: lib/minitest/assertions.rb (hunk 3 above)
Comment (nessamurmur) [inline lib/minitest/assertions.rb:41]: :+1:

**Pull request merged**